app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -125,12 +125,21 @@ def reference_to_hash(reference)
 
     def qualifications
       {
-        gcses: application_form.application_qualifications.gcses.map do |gcse| qualification_to_hash(gcse) end,
-        degrees: application_form.application_qualifications.degrees.map do |degree| qualification_to_hash(degree) end,
-        other_qualifications: application_form.application_qualifications.other.map do |other| qualification_to_hash(other) end,
+        gcses: qualifications_of_level('gcse').map { |q| qualification_to_hash(q) },
+        degrees: qualifications_of_level('degree').map { |q| qualification_to_hash(q) },
+        other_qualifications: qualifications_of_level('other').map { |q| qualification_to_hash(q) },
       }
     end
 
+    def qualifications_of_level(level)
```

Comment:
Would it make sense to replace the existing scopes in `ApplicationQualification` by this? That would mean the other uses if it in the candidate interface would be fixed as well. I can't think of a situation where it would make it slower.

Reply:
I'm not sure about that, because doing it this way transforms the AR relation into an array - so you can't return a relation to do relation chaining (e.g. `application_qualifications.gcses.where(...)` etc)

This approach makes total sense in our case - looping round a relation to render it as json - but probably wouldn't in others. If there are other parts of the app where this N+1 query issue is a problem, point me at them and i'll happily take a look and see what can be done to make the solution transferrable, but I think this stands alone for now

Reply:
Cool!